### PR TITLE
feat(#528): Do not require the Docker host configuration on TestcontainersSettings initialization

### PR DESCRIPTION
**Pull request by HofmeisterAn**
@Xitric @BenasB what do you think about this?

### CODE REVIEW

File: src/Testcontainers/Builders/ImageFromDockerfileBuilder.cs [hunk 1]
```diff
@@ -75,6 +75,9 @@ public IImageFromDockerfileBuilder WithBuildArgument(string name, string value)
     /// <inheritdoc />
     public Task<string> Build()
     {
+      _ = Guard.Argument(this.DockerResourceConfiguration.DockerEndpointAuthConfig, nameof(IDockerResourceConfiguration.DockerEndpointAuthConfig))
```
Comment (BenasB) [inline src/Testcontainers/Builders/ImageFromDockerfileBuilder.cs:78]: One small question: why discard assignment here and not just calling the function alone? that is why `_ =` at the front?

**Review by BenasB — APPROVED** [review]
It seems good. If I understand correctly, we allow `DockerEndpointAuthConfig` to become null at the start, but then enforce it to have a value in the end when building (it has to come from somewhere) - this makes sense. I don't think I can test this on my actual setup because we have limited access to nuget packages (until it is released officially).

P.S. I haven't had the chance to contribute to open source projects yet, so I'm not much of an expert.

File: src/Testcontainers/Builders/ImageFromDockerfileBuilder.cs (hunk 1 above)
Comment (vlaskal) [reply src/Testcontainers/Builders/ImageFromDockerfileBuilder.cs:78]: If I undersand correctly it is new best practice to show that the result is useless for reader or even for compiler. Zou can find out more here: https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/functional/discards 
I would like to know if there is some real compiler optimiyation or is not.

Comment (HofmeisterAn) [reply src/Testcontainers/Builders/ImageFromDockerfileBuilder.cs:78]: As far as I know IL is the same (no compiler optimization).

**Comment by HofmeisterAn** [comment]
ab7fbb4171c1a51e69697c0d9ea39e09529469ac, closes #528, #529.

**Pull request merged**